### ISSUE TITLE
Discuss how to do validation of models

### ISSUE THREAD

**Issue by thelgevold** [jpdonaghy/angular-dynamic-forms#7]
We should discuss how to do validation in our form.

@olore has started this work in a separate branch, so let's discuss how to integrate this with the rest.

The form api does support integration of validators in the form group, so we can look at that too.
I have been playing with that a bit here: http://www.syntaxsuccess.com/viewarticle/forms-and-validation-in-angular-2.0

```
function zipValidator(zip) {
    var valid = /^\d{5}$/.test(zip.value);
    if(valid){
        return null;
    }
    return {"invalidZip":true};
}
this.form = fb.group({
        "firstName": ['', Validators.required],
        "streetAddress": ['',Validators.required],
        "zip": ['', Validators.compose([zipValidator])],
        "type": ['home']
    });
```

There are a few built in validators as well as an api for creating custom ones.

**Comment by jpdonaghy** [comment]
@olore Put the validators in their own folder like src/app/validators, also one thing per file.

**Comment by jpdonaghy** [comment]
We'll need a way to add/remove validators on the fly as form context changes.  For instance on an address form when the user changes the country selection we need to change the validation on the Postal Code field to match the new country context.

**Comment by olore** [comment]
@thelgevold cool, thanks I will take a look. 
@jpdonaghy definitely, I like moving the non angular stuff out and agree 100% one thing per file, that's why it's not a PR yet :)

Validators on the fly sounds like a fun problem to solve

**Comment by thelgevold** [comment]
Yeah, in the case of a formGroup implementation I was picturing the group would be regenerated with new validators based on the request for the new country context
